src/main/fc/tasks.c

```diff
@@ -423,7 +423,7 @@ task_t *getTask(unsigned taskId)
 void tasksInit(void)
 {
     for (int i = 0; i < TASK_COUNT; i++) {
-        tasks[i].attr = &task_attrs[i];
+        tasks[i].attributes = task_attributes[i];
```

Comment:
Hmm. seems that GCC insists on unrolling the struct assignment. This will probably reduce code size:
```suggestion
        memcpy_fn(tasks[i].attributes, task_attributes[i], sizeof(tasks[i].attributes);
```

Reply:
Gives:
```
./src/main/common/utils.h:110:8: note: expected 'const void *' but argument is of type 'task_attributes_t' {aka 'const struct <anonymous>'}
  110 | void * memcpy_fn ( void * destination, const void * source, size_t num ) asm("memcpy");
      |        ^~~~~~~~~
./src/main/fc/tasks.c:426:88: error: expected ';' before '}' token
  426 |         memcpy_fn(tasks[i].attributes, task_attributes[i], sizeof(tasks[i].attributes);
      |                                                                                        ^
      |                                                                                        ;
  427 |     }
      |     ~
```

Reply:
Output:
```
Linking STM32F411 
Memory region         Used Size  Region Size  %age Used
           FLASH:        8072 B        10 KB     78.83%
FLASH_CUSTOM_DEFAULTS:           8 B         6 KB      0.13%
    FLASH_CONFIG:          0 GB        16 KB      0.00%
          FLASH1:      487431 B       480 KB     99.17%
FLASH_CUSTOM_DEFAULTS_EXTENDED:          0 GB         0 GB
   SYSTEM_MEMORY:          0 GB        29 KB      0.00%
             RAM:      112136 B       128 KB     85.55%
       MEMORY_B1:          0 GB         0 GB
   text	   data	    bss	    dec	    hex	filename
 489911	   5600	 106136	 601647	  92e2f	./obj/main/betaflight_STM32F411.elf
```

Reply:
Hmm .. suing `memcpy` has minimal effect:
```
.L1003:
	mov	lr, r5
	ldmia	lr!, {r0, r1, r2, r3}
	mov	ip, r6
	stmia	ip!, {r0, r1, r2, r3}
	ldm	lr, {r0, r1}
	adds	r5, r5, #24
	cmp	r8, r5
	stm	ip, {r0, r1}
	add	r6, r6, #80
	bne	.L1003
```
vs
```
.L1003:
	mov	r1, r6
	mov	r0, r5
	movs	r2, #24
	adds	r5, r5, #80
	bl	memcpy
	cmp	r7, r5
	add	r6, r6, #24
	bne	.L1003
```

Reply:
Master seems to be the best solution according to these tables.Let's stick with renaming only.

Reply:
I just don't get where is size difference coming from. Compiler is saving 1 instruction on attributes access. But maybe 32bit instruction encoding has to be used with larger 'taska', making code larger ...

Reply:
Another possibility is that with smaller code, gcc decides to unroll / inline function somewhere ...

Reply:
No worries - at least we tried